packages/core/src/NodeExecuteFunctions.ts

```diff
@@ -508,24 +508,50 @@ async function proxyRequestToAxios(
 	});
 
 	if (configObject.auth?.sendImmediately === false) {
+		// for digest-auth
 		const { auth } = axiosConfig;
 		delete axiosConfig.auth;
-		const header = await axios(axiosConfig).catch((error) => {
+		const resp1 = await axios(axiosConfig).catch((error) => {
 			if (error.response.status === 401) {
-				return error.response.headers['www-authenticate'];
+				return error;
 			}
 			throw error;
 		});
-		const userpass = `${auth?.username as string}:${auth?.password as string}`;
-		axiosConfig.headers = {};
-		axiosConfig.headers.Authorization = digest(
-			axiosConfig.method,
-			axiosConfig.url,
-			header,
-			userpass,
-		);
-		axiosConfig.headers.Authorization += ', algorithm=MD5';
+		const authDetails = resp1.response.headers['www-authenticate']
```

Comment:
We have to check the contents of `resp1` and see if response is available and also if `www-authenticate` header is present. Otherwise, throw the same error. This can be done inside the above `catch` block.